core/lib/eth_client/src/lib.rs

```diff
@@ -76,14 +79,14 @@ impl Options {
 pub trait EthInterface: Sync + Send {
     /// Fetches the L1 chain ID (in contrast to [`BoundEthInterface::chain_id()`] which returns
     /// the *expected* L1 chain ID).
-    async fn fetch_chain_id(&self) -> Result<L1ChainId, Error>;
+    async fn fetch_chain_id(&self) -> EnrichedClientResult<L1ChainId>;
```

Comment:
If necessary, I can extract these changes to `eth_client` ([refactoring its error types](https://github.com/matter-labs/zksync-era/pull/2000/commits/26241de8eb3cc47fdba2069547b130724e973da6)) in a separate PR. It allowed to simplify error handling for the L1 data fetcher.